AutoGrade/grader.py

```diff
@@ -90,13 +90,14 @@ def run_student_tests(target_folder, total_points, timeout):
     p.start()
     p.join(timeout + 1) # Pytest will also timeout
 
+    timeout = False
     #In case process is stuck in infinite loop or something
     if p.is_alive():
```

Comment:
I also think that we will have to remove that join now because pytest-timeout assign timeout for each test case separately.

**a01.py** (Check sleep function)
```
from time import sleep

def code_holder(num1, num2):
    v = 0 
    v = num1 + num2
    sleep(2.5)
    return v 
```
**test-student.py** (timeout 3 seconds while execution will take 5 seconds. Result is 2 passed)
```
from a01 import code_holder

def test_answer():
    assert code_holder(1, 2) == 3

def test_second():
    assert code_holder(0, 0) == 0 
```
@recluze I tried to reproduced that case where `pytest-timeout` fail single test case but in my scenario pytest is killing the whole process when single timeout occur.

**a01.py**
```
from time import sleep

def code_holder(num1, num2):
    v = 0 
    v = num1 + num2
    sleep(2) # This case should pass
    return v 

def code_holder1(num1, num2):
    v = 0 
    v = num1 + num2
    sleep(4) # This one make the pytest to break the whole execution
    return v 
```

**test-students.py**
```
from a01 import code_holder, code_holder1 

def test_answer():
    assert code_holder(1, 2) == 3

def test_second():
    assert code_holder1(0, 0) == 0 
```

Reply:
So, what's the conclusion? Does it fail only one test for you? It does fail just the one bad test for me and everything else (as well as the final result) works fine for everything else.

Reply:
Yep, just tried your last example and the first one passes while second fails. So, I guess we're good with this. If you agree, just remove this current PR and remove the timeout from the join. Don't think we need to remove the join, just the timeout param removed from join should do the trick.

Reply:
Changed the order to run the bad test case first. It fails and the second one passes (as it should).

Reply:
It is killing the whole execution for me in second example. Are you testing it using `python run.py local`?

Reply:
No. I'm doing `pytest --timeout=3`. Shouldn't make a difference though ... That would be bad.

Reply:
Same with `run.py local`. Got this : `test_student.py F. ` ... so, first failed but second was run fine.

Reply:
hmmm... getting same result through that as well. 

Well leave that, I will update the code to use pytest-timeout...